### PR TITLE
Unify merge predicate

### PR DESCRIPTION
**Pull request by tavplubix**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
This is an automated comment for commit 356eec1aef9c7872a5c2aa058072555f1e0696a5 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51344/356eec1aef9c7872a5c2aa058072555f1e0696a5/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Daemon/BaseDaemon.cpp [hunk 1]
```diff
@@ -418,7 +418,7 @@ class SignalListener : public Poco::Runnable
             SentryWriter::onFault(sig, error_message, stack_trace);
 
             /// Advice the user to send it manually.
-            if constexpr (std::string_view(VERSION_OFFICIAL).contains("official build"))
+            if (String(VERSION_OFFICIAL).contains("official build"))
```
Comment (alexey-milovidov) [inline src/Daemon/BaseDaemon.cpp:421]: Extra allocation in the fatal signal handler is not good.

Comment (tavplubix) [reply src/Daemon/BaseDaemon.cpp:421]: There are a lot of allocations on strings formatting and logging, so I don't think it's critical. But I committed this change by mistake, it was to fix the "unreachable code" error when building locally

**Comment by tavplubix** [comment]
Integration tests (asan) [4/6] - https://github.com/ClickHouse/ClickHouse/issues/50309

**Pull request merged**